### PR TITLE
CI: restore npdev build

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: ci/azure/posix.yml [hunk 1]
```diff
@@ -44,15 +44,13 @@ jobs:
...
-        # Disabled for NumPy object-dtype warning.
-        # https://github.com/pandas-dev/pandas/issues/30043
-        # py37_np_dev:
-        #   ENV_FILE: ci/deps/azure-37-numpydev.yaml
-        #   CONDA_PY: "37"
-        #   PATTERN: "not slow and not network"
-        #   TEST_ARGS: "-W error"
-        #   PANDAS_TESTING_MODE: "deprecate"
-        #   EXTRA_APT: "xsel"
+         py37_np_dev:
```
Comment (TomAugspurger) [inline ci/azure/posix.yml]: Perhaps indented one space too far.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
So something was reverted on the numpy side for this right? Nothing required yet on our end?

**Comment by jbrockmendel** [comment]
> So something was reverted on the numpy side for this right? Nothing required yet on our end?

I think they reverted the implicit-object warning.

Also I think npdev is now 1.19, so we may need a separate 1.18 build

**Comment by WillAyd** [comment]
Thanks!

**Pull request merged**